### DISCUSSION TITLE
post/migrating-to-isolated-durable-functions

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#7] [Announcements]
# post/migrating-to-isolated-durable-functions

Mark Heath's Development Blog

https://markheath.net/post/migrating-to-isolated-durable-functions

**Comment by Mech0z** [comment]
Structured logging seems none trivial to get working, we have had problems since going to isolated functions in .NET 6 
Our issue https://github.com/Energinet-DataHub/geh-charges/issues/1882 and https://github.com/Azure/azure-functions-dotnet-worker/issues/423#issuecomment-1362811736

**Reply by markheath** [reply]
thanks, that's useful to know

**Comment by bzbetty** [comment]
re: Newtonsoft. 
Add a reference to nuget Microsoft.Azure.Core.NewtonsoftJson, then in program.cs use

  .ConfigureFunctionsWorkerDefaults(x =>
    {
        x.UseNewtonsoftJson();
    })

**Reply by markheath** [reply]
thanks - that's a good tip for migration scenarios where you don't want to change existing behaviour

**Comment by steve-torchia** [comment]
Great article, Thanks so much!  I'm watching this space closely as we currently have a fair number of durable functions in production.   Right now, there does not seem to be any justifiable reason to migrate given the amount of work it would take.   Have you come across any comments or articles talking about making the migration easier?  

thanks again!

**Reply by markheath** [reply]
well when I started this I thought I might write a migration guide, but quickly realized that it would be very long as so many things have changed. I agree that it doesn't really make sense to migrate a project at this stage. It might become a bit easier when SDK type support is added.

**Comment by sundeep22** [comment]
Great article. I have a big project coming my way that needs to be built using durable functions. I was wondering why .Net 7 Isolated process wasn't showing the durable functions template in VS 2022.. so it looks like they haven't released it yet! I hope they do that soon.. I don't want to start off in dotnet 6 and having to migrate it!

**Comment by Yogeswaran66** [comment]
Hi. Can you show how to return a List of DTO from activity function. I managed to return a DTO from activity function turns out it's not easy as you have to put the DTO inside a specific namespace. But List<dto> is not working. Can you show a demo for that

**Reply by markheath** [reply]
I'm a bit surprised that wouldn't work. I'll give it a try next time I so some work on this

**Reply by Yogeswaran66** [reply]
Thanks. It works 
`
var readDtoTasks = new List<Task<List<MyDTO>>>();

foreach (var items in chunkedLeadIds)
{
	readDtoTasks.Add(context.CallActivityAsync<List<MyDTO>>("MyActivityFunc", items));
}

await Task.WhenAll(readDtoTasks);
var items = readDtoTasks.SelectMany(x => x.Result).ToList();
`

**Comment by andsj073** [comment]
Thanks again for a great post Mark!
Speaking of Durable Functions, have you ever tried to, or heard of anyone, use SQL Database as TaskHub provider _AND_ configuring Column Encryption on for example the Payloads.Text column that could have sensitive data? 
On paper it should work one might think, given the Function App has the right Key Vault role assignment, and the NuGet SqlColumnEncryptionAzureKeyVaultProvider is included. 
But does it in practice? Any insights?

**Reply by markheath** [reply]
that's a great question, but not something I've tried myself. probably a question to ask the Durable Functions team on GitHub

**Comment by mdddev** [comment]
Hi Mark, at the end of your post you're referring to the new strongly-typed class-based syntax of writing durable functions code. You even link to the official sample. **Have you succeeded in running this sample?** In my testing, I have failed and opened an [Github issue](https://github.com/microsoft/durabletask-dotnet/issues/136) for it. So I am eager to know other's experiences with it.

**Reply by markheath** [reply]
I don't recall trying it when I wrote this post. I just noticed it and thought it was interesting. I've subscribed to that GitHub issue you linked as it will be interesting to see where this goes.

**Comment by DBJDBJ** [comment]
I am using nodes js from containers, running on Kubernetes. I find that feasible. No, I am not a troll.
